### PR TITLE
Improve a bit some regexp

### PR DESCRIPTION
**Pull request by jvoisin**
- Prepend `^` when possible
- Append `$` when possible

Those changes have been tested on official websites of changed
technologies.

### CODE REVIEW

File: src/apps.json [hunk 1]
```diff
@@ -551,7 +551,7 @@
 				"22"
 			],
 			"headers": {
-				"Server": "(?:Apache(?:$|/([\\d.]+)|[^/-])|(?:^|\b)HTTPD)\\;version:\\1"
+				"Server": "^(?:Apache(?:$|/([\\d.]+)|[^/-])|(?:^|\b)HTTPD)\\;version:\\1"
```
Comment (gadcam) [inline src/apps.json]: Are you sure that Apache is always the first thing to appear in the header ? If I am right when several apps are adding content to the header it is not true.

File: src/apps.json [hunk 2]
```diff
@@ -1800,8 +1800,8 @@
 				"1"
 			],
 			"headers": {
-				"Set-Cookie": "CraftSessionId=",
-				"X-Powered-By": "Craft CMS"
+				"Set-Cookie": "^CraftSessionId=",
```
Comment (gadcam) [inline src/apps.json]: Same remark : when several cookies are set at the same time will this rule work ?

Comment (jvoisin) [reply src/apps.json]: How would this look like? Can you actually set several cookies inside a single `Set-Cookie` header?

Comment (AliasIO) [reply src/apps.json]: Cookies are separated by semi-colons in a single header.

Comment (jvoisin) [reply src/apps.json]:
```
jvoisin@mim 12:22 ~ curl https://www.facebook.com -I
HTTP/1.1 200 OK
X-XSS-Protection: 0
Pragma: no-cache
Cache-Control: private, no-cache, no-store, must-revalidate
X-Frame-Options: DENY
Strict-Transport-Security: max-age=15552000; preload
X-Content-Type-Options: nosniff
Expires: Sat, 01 Jan 2000 00:00:00 GMT
Set-Cookie: fr=0utGfGF1ho04AU122..BaBYv6.ud.AAA.0.0.BaBYv6.AWUx6yQT; expires=Thu, 08-Feb-2018 11:22:34 GMT; Max-Age=7776000; path=/; domain=.facebook.com; secure; httponly
Set-Cookie: sb=-osFWsw8cvg5m6sNBEE9aN1d; expires=Sun, 10-Nov-2019 11:22:34 GMT; Max-Age=63072000; path=/; domain=.facebook.com; secure; httponly
Vary: Accept-Encoding
Content-Type: text/html; charset=UTF-8
X-FB-Debug: ymZ5XMIeSzwroDHBm1w+vUVGupOvjhtHLTo0qmQvNqqhMj/tu9DNpF0Bjem9tdyHFh8fQAnYCTUfIDUPrkAyeg==
Date: Fri, 10 Nov 2017 11:22:34 GMT
Connection: keep-alive

jvoisin@mim 12:22 ~ 
```

It seems to me that the header `Set-Cookie` is sent twice. Semi-colons are to separate attributes.

Comment (gadcam) [reply src/apps.json]: I don't have a definitive answer but you should look into https://stackoverflow.com/questions/11533867/set-cookie-header-with-multiple-cookies and https://stackoverflow.com/questions/2880047/is-it-possible-to-set-more-than-one-cookie-with-a-single-set-cookie.

Comment (jvoisin) [reply src/apps.json]: Apparently, it's deprecated, I don't know if we want to support this syntax :/

Comment (gadcam) [reply src/apps.json]: Moving on https://developer.mozilla.org/fr/Add-ons/WebExtensions/API/cookies would solve this issue, what do you think ? (and for the NPM driver we could use browser.cookies / browser.cookies.dump(output?))

**Comment by AliasIO** [comment]
Thanks!

**Pull request merged**